freeglut/freeglut/src/fg_menu.c

```diff
@@ -780,14 +780,29 @@ void fghCalculateMenuBoxSize( void )
 /*
  * Creates a new menu object, adding it to the freeglut structure
  */
-int FGAPIENTRY glutCreateMenu( FGCBMenu callback )
+int FGAPIENTRY glutCreateMenuUcall( FGCBMenuUC callback, FGCBUserData userData )
 {
     /* The menu object creation code resides in fg_structure.c */
-    FREEGLUT_EXIT_IF_NOT_INITIALISED ( "glutCreateMenu" );
+    FREEGLUT_EXIT_IF_NOT_INITIALISED ( "glutCreateMenuUcall" );
     if (fgState.ActiveMenus)
         fgError("Menu manipulation not allowed while menus in use.");
 
-    return fgCreateMenu( callback )->ID;
+    return fgCreateMenu( callback, userData )->ID;
+}
+
+/* Standard glutCreateMenu */
+static void glutCreateMenuCallback( int menu, FGCBUserData userData )
```

Comment:
Thanks! Seems like a sensible approach to make old and new work.
- whether function should be `fg` or `fgh` doesn't really, matter i don't think its consistent in the current codebase anyway. but some prefix like that shows its internal, not API.
- if generalizing becomes such a mess that there is no benefit, then it may not be worth it. Your call!